### PR TITLE
Provide path to local `sentry-cli`

### LINKED ISSUE #220: Allow to set cliPath via options
We are building flutter apps in a Docker container with preinstalled sentry-cli.

Please allow to set up sentry-cli path from configuration to speed up building and reduce unnecessary network usage.

Also I think that preinstalled binary must be used first and then fallback to downloading.

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Provide the path to a local `sentry-cli` binary.

## :bulb: Motivation and Context

Avoid downloading of sentry-cli by providing own version.

Closes #220

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by denrase** [comment]
@buenaflor My only concern here is that combinations of plugin and sentry-cli versions. We could validate the checksum, but then CI would fail, so I'm not sure if this is better than the current approach. WDYT?

**Review by buenaflor — COMMENTED** [review]
looks good but left some comments

File: lib/src/configuration.dart [hunk 1]
```diff
@@ -206,27 +211,39 @@ class Configuration {
...
+      if (platform != null) {
+        await injector.get<CLISetup>().check(platform, binPath);
+      } else {
+        Log.warn(
+            'Host platform not supported. Cannot verify sentry-cli checksum.');
+      }
+      Log.info("Using sentry-cli at path '$binPath'");
+      cliPath = binPath;
+    } else {
+      if (platform == null) {
```
[the remark this answers is not in the document]
Comment (denrase) [reply lib/src/configuration.dart]: I extracted methods & removed side-effects.

File: lib/src/cli/setup.dart [hunk 2]
```diff
@@ -59,12 +68,12 @@ class CLISetup {
     final expected = source.hash;
     if (calculated != expected) {
       Log.warn(
-          "Sentry CLI checksum mismatch on ${file.path} - expected $expected but got $calculated");
+          "Sentry CLI checksum mismatch on ${file.path} - expected $expected but got $calculated.");
```
[the remark this answers is not in the document]
Comment (denrase) [reply lib/src/cli/setup.dart:71]: Added the version and updated the script.

**Review by buenaflor — APPROVED** [review]
👍

**Pull request merged**